nixio/pycore/h5group.py

```diff
@@ -70,34 +94,76 @@ def get_dataset(self, name):
         else:
             raise notfound
 
+    def write_data(self, name, data, dtype=None):
+        """
+        Writes the data to a Dataset contained in the group with the given name.
+        Creates the Dataset if necessary.
+
+        :param name: name of the Dataset object
+        :param data: the data to write
+        :param dtype: optionally specify the data type, otherwise it will be
+        automatically determined by the data
+        """
+        shape = np.shape(data)
+        if self.has_data(name):
+            dset = self.get_dataset(name)
+            dset.shape = shape
+        else:
+            if dtype is None:
+                dtype = DataType.get_dtype(data[0])
+            dset = self.create_dataset(name, shape, dtype)
+
+        dset.write_data(data)
+
     def get_data(self, name):
         """
-        Returns the data contained in the dataset identified by 'name', or None
-        if the a dataset of that name does not exist in the Group.
+        Returns the data contained in the dataset identified by 'name', or an
+        empty list if a dataset of that name does not exist in the Group.
 
         :param name: The name of the dataset
         :return: The data contained in the dataset as a numpy array or None
         """
         if name not in self.group:
-            return tuple()
+            return []
```

Comment:
The same way None is returned when an attribute that doesn't exist is requested in `get_attr`, I thought it was more appropriate to return an empty list here.
Since certain data properties are removed when they're empty, such as SetDimension.labels, I felt it should be the calling class' responsibility to allow empty or missing data.

I think NIX behaves the same way. For instance, in NIX, RangeDimension.ticks() handles the error if the group does not contain a "ticks" or "data" data object. Conversely, SetDimension.labels() returns an empty vector if "labels" is missing.